contracts/modules/VersionManager.sol

```diff
@@ -0,0 +1,313 @@
+// Copyright (C) 2018  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./common/Utils.sol";
+import "../infrastructure/base/Owned.sol";
+import "../infrastructure/storage/ITransferStorage.sol";
+import "../infrastructure/storage/IGuardianStorage.sol";
+import "./common/IModule.sol";
+import "./common/BaseFeature.sol";
+
+/**
+ * @title VersionManager
+ * @notice Intermediate contract between features and wallets. VersionManager checks that a calling feature is
+ * authorised for the wallet and if so, forwards the call to it.
+ * @author Olivier VDB <olivier@argent.xyz>
+ */
+contract VersionManager is IVersionManager, IModule, BaseFeature, Owned {
+
+    bytes32 constant NAME = "VersionManager";
+
+    bytes4 constant internal ADD_MODULE_PREFIX = bytes4(keccak256("addModule(address,address)"));
+    bytes4 constant internal UPGRADE_WALLET_PREFIX = bytes4(keccak256("upgradeWallet(address,uint256)"));
+
+    // Last bundle version
+    uint256 public lastVersion;
+    // Version to be used for new wallet creation
+    uint256 public newWalletVersion = 1;
```

Comment:
At wallet creation we will always use latest version (feature set)? So with create2 we will no longer include the list of modules (or even a version number) when generating the counterfactual addr?

Makes sense from cost reduction perspective.

Reply:
At wallet creation time, the version used to initialise the wallet is `newWalletVersion` which can be changed by the multisig. This is so that, in the scenario discussed in the common chat with Sergio (where the client does not yet support the latest on-chain version), the new wallet can be created with a version compatible with the client.

The downside (compared to 2.0) is that this new 2.1 architecture introduces some trust: the user has to trust that a new faulty bundle will not be released by the time the wallet is created. An alternative could be to create all wallets on V1 and require the user to do the upgrade from V1 to the version supported by the client, but it's going to be more costly.

Reply:
@olivdb you're changing this yes? I think the guys will have discussed it with you this morning so that wallets are created with a specific version, and not `newWalletVersion` ?

Reply:
Yes, it's changed as of 47b95a3.